src/Compat/SearchControls.php

```diff
@@ -83,7 +86,7 @@ public function createSearchBar(Query $query, ...$params): SearchBar
         if (method_exists($this, 'completeAction')) {
             $searchBar->setSuggestionUrl(Url::fromPath(
                 "$moduleName/$controllerName/complete",
-                ['_disableLayout' => true, 'showCompact' => true]
+                array_merge($paramsToAdd, ['_disableLayout' => true, 'showCompact' => true])
```

Comment:
Since you will always have a UrlParams object after not transforming them always to an array, you can just call `$paramsToAdd->set(…)` twice.